vscode/src/azure/networkRequests.ts

```diff
@@ -0,0 +1,291 @@
+  }
+
+  // Needs to POST an application/json payload such as: {"containerName": "job-073064ed-2a47-11ee-b8e7-010101010000","blobName":"outputData"}
+  sasUri() {
+    return `${this.endpoint}${this.id}/storage/sasUri?api-version=${this.apiVersion}`;
+  }
+}
+
+export class StorageUris {
+  // Requests use a Shared Access Signature. See https://learn.microsoft.com/en-us/rest/api/storageservices/service-sas-examples
+
+  // x-ms-date header should be present in format: Sun, 06 Nov 1994 08:49:37 GMT
+  // See https://learn.microsoft.com/en-us/rest/api/storageservices/representation-of-date-time-values-in-headers
+
+  readonly apiVersion = "2023-01-03"; // Pass as x-ms-version header (see https://learn.microsoft.com/en-us/rest/api/storageservices/versioning-for-the-azure-storage-services#authorize-requests-by-using-azure-ad-shared-key-or-shared-key-lite)
+
+  // Same for PUT, with a status code of 201 if successful
+  getContainer(storageAccount: string, container: string, sas: string) {
+    return `https://${storageAccount}.blob.core.windows.net/${container}?restype=container&${sas}`;
+  }
+
+  // Same for DELETE, with a status code of 202 if successful
+  // Also same URI for PUT, but must include the following headers:
+  // - x-ms-blob-type: BlockBlob
+  // - Content-Length: <n>
+  // It will return 201 if created.
+  getBlob(
+    storageAccount: string,
+    container: string,
+    blob: string,
+    sas: string
+  ) {
+    return `https://${storageAccount}.blob.core.windows.net/${container}/${blob}?${sas}`;
+  }
+}
+
+export async function compileToBitcode(
+  compilerService: string,
+  providerId: string,
+  qir: string
+) {
+  try {
+    log.info("Using compiler service at " + compilerService);
+    const bitcodeRequest = await fetch(compilerService, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/octet-stream",
+        "x-hardware-target": providerId,
+      },
+      body: qir,
+    });
+
+    if (!bitcodeRequest.ok) {
+      log.error("Failed to compile to QIR bitcode", bitcodeRequest);
+      throw Error("Failed to compile to QIR bitcode");
+    }
+    return new Uint8Array(await bitcodeRequest.arrayBuffer());
+  } catch (e) {
+    log.error("Failed to compile to QIR bitcode", e);
+    throw Error("Failed to compile to QIR bitcode");
+  }
+}
+
+// Put all the Response types in a namespace for easy importing
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+export namespace ResponseTypes {
+  export type Tenant = {
+    id: string;
+    tenantId: string;
+    displayName: string;
+  };
+
+  export type TenantList = {
+    value: Array<Tenant>;
+  };
+
+  export type Subscription = {
+    id: string;
+    subscriptionId: string;
+    tenantId: string;
+    displayName: string;
+  };
+
+  export type SubscriptionList = {
+    value: Array<Subscription>;
+  };
+
+  export type Provider = {
+    providerId: string; // e.g., 'ionq', 'quantinuum', 'rigetti'
```

Comment:
Remove partner IDs from code
```suggestion
    providerId: string;
```